scripts/core/editor3/components/links/AttachmentList.tsx

```diff
@@ -22,38 +24,47 @@ export class AttachmentList extends React.Component<any, any> {
 
-    render() {
-        if (!this.state.attachments) {
-            return <div />; // wait for attachments
+    renderAttachment(attachment: IAttachment) {
+        if (attachment.internal === true) {
+            return null;
         }
 
-        const attachments = this.state.attachments.map((file) => {
-            const fileicon = 'big-icon--' + FileiconFilter()(file.mimetype);
-            const filesize = FilesizeFilter()(file.media.length);
+        const fileicon = 'big-icon--' + FileiconFilter()(attachment.mimetype);
+        const filesize = FilesizeFilter()(attachment.media.length);
 
-            const className = classNames('sd-list-item', {
-                'sd-list-item--active': file._id === this.props.selected,
-            });
+        const className = classNames('sd-list-item', {
+            'sd-list-item--active': attachment._id === this.props.selected,
+        });
 
-            return (
-                <div key={file._id} className={className} onClick={() => this.props.onClick(file)}>
-                    <div className="sd-list-item__column sd-list-item__column--no-border">
-                        <i className={fileicon} />
+        return (
+            <div key={attachment._id} className={className} onClick={() => this.props.onClick(attachment)}>
+                <div className="sd-list-item__column sd-list-item__column--no-border">
+                    <i className={fileicon} />
+                </div>
+                <div className="sd-list-item__column sd-list-item__column--grow">
+                    <div className="sd-list-item__row">
+                        <h4>{attachment.title}</h4>
                     </div>
-                    <div className="sd-list-item__column sd-list-item__column--grow">
-                        <div className="sd-list-item__row">
-                            <h4>{file.title}</h4>
-                        </div>
-                        <div className="sd-list-item__row">
-                            <h5>{file.filename} ({filesize})</h5>
-                        </div>
-                        <div className="sd-list-item__row">
-                            <div className="description">{file.description}</div>
-                        </div>
+                    <div className="sd-list-item__row">
+                        <h5>{attachment.filename} ({filesize})</h5>
+                    </div>
+                    <div className="sd-list-item__row">
+                        <div className="description">{attachment.description}</div>
+                        <div className="description">{attachment.description}</div>
                     </div>
                 </div>
-            );
-        });
+            </div>
+        );
+    }
+
+    render() {
+        if (!this.state.attachments) {
+            return <div />; // wait for attachments
+        }
+
+        const attachments = this.state.attachments
+            .map(this.renderAttachment)
+            .filter(Boolean);
```

Comment:
Don't use implicit casting. When you have `if (x)` it's clearer when it is written as `if (x != null)` if that's what you mean.

I would avoid filtering on data rather than UI components here:

```
render() {
    if (this.state.attachments != null) {
        return null; // wait until loaded
    }

    const externalAttachements = this.state.attachments.filter((attachment) => attachment.internal !== true);

    if (externalAttachements.length > 0) {
        return (
            <div>
                {externalAttachements.map((attachment, i) => {
                    <AttachmentItem attachment={attachment} />
                })}
            </div>
        );

    } else {
        return (
            <p style={{padding: 20, margin: 0}}>
                {gettext('There are no attachments yet. Upload some first using Attachments widget.')}
            </p>
        );
    }
}
```

Reply:
ha, I didn't read this comment but I've already done something similar on my latest commit